catalog/solr/catalog-solr-solrclient/src/main/java/ddf/catalog/source/solr/rest/SolrRest.java

```diff
@@ -0,0 +1,154 @@
+    }
+
+    public void setB(Float b) {
+        LOGGER.trace("Setting B property: {}", b);
+        this.b = b;
+    }
+
+    public void setSolrSchemaUrl(String solrSchemaUrl) {
+        this.solrSchemaUrl = solrSchemaUrl;
+    }
+
+    public Float getK1() {
+        return k1;
+    }
+
+    public Float getB() {
+        return b;
+    }
+
+    public String getSolrSchemaUrl() {
+        return solrSchemaUrl;
+    }
+
+    public void getProperties() {
+        try {
+            String response = factory.getClient()
+                    .getFieldTypes("json");
+
+            Map<String, Object> map = gson.fromJson(response,
+                    new TypeToken<Map<String, Object>>() {}.getType());
+
+            fieldTypes = (ArrayList<Object>) map.get("fieldTypes");
+        } catch (Exception e) {
+            LOGGER.debug("Unable to getProperties from: {}", solrSchemaUrl, e);
+        }
+    }
+
+    public void init() {
+        if (solrSchemaUrl != null) {
+            factory = new SecureCxfClientFactory<>(solrSchemaUrl, SolrRestClient.class);
+            similarityFormat = new LinkedTreeMap<>();
+            similarityFormat.put("class", "solr.BM25SimilarityFactory");
+            similarityFormat.put(K1_PROPERTY, k1);
+            similarityFormat.put(B_PROPERTY, b);
+
+            getProperties();
+            setSimilarities();
+        }
+    }
+
+    public void refresh(Map<String, Object> properties) {
+        if (MapUtils.isEmpty(properties)) {
+            LOGGER.debug("No properties specified during refresh.");
+            return;
+        }
+
+        LOGGER.trace("Refresh configuration with properties: {}", properties);
+
+        Object solrUrlObject = properties.get(SOLR_URL_PROPERTY);
+        if (solrUrlObject instanceof String) {
+            setSolrSchemaUrl((String) solrUrlObject);
+        }
+
+        Object k1Object = properties.get(K1_PROPERTY);
+        if (k1Object instanceof Float) {
+            setK1((Float) k1Object);
+        }
+
+        Object bObject = properties.get(B_PROPERTY);
+        if (bObject instanceof Float) {
+            setB((Float) bObject);
+        }
+
+        init();
+    }
+
+    private void setSimilarities() {
+        if (CollectionUtils.isNotEmpty(fieldTypes)) {
+            for (Object fieldType : fieldTypes) {
+                LinkedTreeMap<String, Object> objectLinkedTreeMap =
+                        (LinkedTreeMap<String, Object>) fieldType;
+                objectLinkedTreeMap.put("similarity", similarityFormat);
+                LinkedTreeMap<String, Object> replaceField = new LinkedTreeMap<>();
+                replaceField.put("replace-field-type", objectLinkedTreeMap);
+
+                LOGGER.trace("Replacing field: {}", gson.toJson(replaceField));
+
+                String response = factory.getClient()
+                        .replaceField(gson.toJson(replaceField));
+
+                LOGGER.trace("Configuration update response: {}", response);
```

Comment:
❗️ The same applies here. You're not doing anything with the `response` but logging it at trace level; the trace should be wrapped in a level check.

Reply:
Actually after further investigation we still need to make the call to replace the field, so there isn't too much gained wrapping that LOGGER statement in a check.

Reply:
Fixed the previous to check log level.